### DISCUSSION TITLE
Profiling Tool for NextJS Applications

### DISCUSSION THREAD

**Discussion by binoiii** [vercel/next.js#65567] [Help]
### Summary

We want to include in our workflow where we measure the performance and profile our application prior to deployment so we won't introduce any regression related to performance.

Scenario
1. Dev1 developed a new feature i.e` feature-1`.
2. Dev1 will create a feature environment that has `feature-1`.
3. Dev1 should be able to do profiling to determine that `feature-1` won't introduce performance regression. 
4. If performance is within accepted values, merge branch with `feature-1` to DEV.
5. If it is time to release, merge DEV to PROD and deploy PROD.

What tool would you recommend that is best fit to NextJS application?

Thanks in advance!

**Comment by JaimeenMakavana** [comment]
if you deploy your application on vercel then it is providing you analytics tools through which you can measure performance and profile also.

![image](https://github.com/vercel/next.js/assets/129974169/7fe6661e-97b4-48da-aa82-5ff6340e85a7)

**Reply by binoiii** [reply]
Ahhh outside Vercel, do have any in mid to recommend?

**Reply by SouthLink** [reply]
[lighthouse](https://github.com/GoogleChrome/lighthouse) and [sentry](https://github.com/getsentry/sentry) are recommended
